api/app/uploaders/attachment_uploader.rb

```diff
@@ -49,6 +50,10 @@ class AttachmentUploader < Shrine
     outcome
   end
 
+  add_metadata :alt_text do |io, _context|
+    io.respond_to?(:alt_text) ? io.alt_text : nil
```

Comment:
it's a very minor thing, but you could do `io.try(:alt_text)` to do this with the same result, but more succinct and idiomatic rails

Reply:
Also while we're not currently using it, I wanted to go ahead and check to make sure that if we end up adding `refresh_metadata` plugin, it won't break anything. Because `add_metadata` defines a reader on the UploadedFile, and `io` when using refresh metadata is the `UploadedFile` instance, it will just transparently work with the way the block is written. For new uploads, `io.alt_text` will be reading from the data file. For existing uploads, when refreshing, it will be the previous alt text value and we won't lose it when refreshing.

Reply:
Just making sure I'm clear, you already checked `refresh_metadata` and we're good? Fwiw, I think I came across a related issue making `IngestionSources` uploadable from the client: https://github.com/ManifoldScholar/manifold/pull/3597/files#r1237821815

Reply:
Yeah, I pulled your branch down and tested adding refresh_metadata just to make sure the alt text was persistent. Not that we need to add it now, but I wasn't 100% on how shrine would behave with ephemeral metadata like this.